### ISSUE TITLE
How does processing a Stripe payment work?

### ISSUE THREAD

**Issue by anomitra** [mirumee/django-payments#143]
Hi, I have been trying to use `django-payments` to implement a Stripe-based checkout flow. I can render the input form for Stripe details, using `payments.get_form()`, but after that, how is the payment processed? I can see that the outgoing call is sent to Stripe -- however, I do not understand how that change is supposed to be reflected at my application's back-end.

There is no `process_data` implementation for `StripeProvider` as well. I can see that in the `StripeFormMixin.save()` method the Payment status is being set to `PREAUTH` and the transaction_id is generated (which should be enough to capture the payment), however, I cannot understand where or when it is called. Can anyone help? Thanks in advance!

**Comment by amorphic** [comment]
@anomitra did you get any further with this? Once #144 is resolved we're planning on using Stripe also...

**Comment by anomitra** [comment]
@amorphic no, I didn't. I had to stop working on this for several reasons. If you do plan to work on this, let me know if I can be of any help.

**Comment by amorphic** [comment]
@anomitra I did ultimately get this working for our [web shop](https://shop.enstaved.com).

I had to hack together a fix to #144 because Saleor (which we're using) was updated to Django 1.11 and that broke the `StripreCheckoutWidget` which is the one Saleor implements via `StripeProvider` -> `ModalPaymentForm`. I'm going to submit this as a PR in the next day or two.

I agree that the code can be hard to follow. It would be really nice if the various methods at least had some basic docstrings to explain what they do.

There's some info the Stripe checkout docs [here](https://stripe.com/docs/checkout) that might help you understand how it works in the context.

**Comment by amorphic** [comment]
@anomitra it turns out that @elwoodxblues beat me to the punch and has cut a new release with Django 1.11 compatibility!

Stripe payments definitely work as outlined in my comment above so I think this can be closed off now?

**Comment by anomitra** [comment]
> a new release with Django 1.11 compatibility!

That's great news!

> Stripe payments definitely work

I wouldn't know about that since I stopped pursuing this, but since you've gotten it to work so it's definitely possible. I'm okay with closing this _(although as you said, some documentation would be most welcome!)_
